### PR TITLE
Add setCWD() method to the System class.

### PR DESCRIPTION
**Pull request by agathoklisx**
Enhance System with setCWD() to set current working directory.

### CODE REVIEW

**Comment by Jason2605** [comment]
Thanks for the PR @agathoklisx! Seems as though the CI/CD integration is failing because it's missing the `setCWD.du` test file.

**Comment by agathoklisx** [comment]
On Mon, Mar 02, at 06:09 Jason_000 wrote:
> Thanks for the PR @agathoklisx! Seems as though the CI/CD integration is failing because it's missing the `setCWD.du` test file.

Sorry, i didn't `git add' in my local copy.

Done, but in the PR slipped in other changes too (as i view the diff from githun) and
i don't know if it is normal, so please check first before apply.

**Comment by agathoklisx** [comment]
On Mon, Mar 02, at 06:20 agathoklisx wrote:
> On Mon, Mar 02, at 06:09 Jason_000 wrote:
> > Thanks for the PR @agathoklisx! Seems as though the CI/CD integration is failing because it's missing the `setCWD.du` test file.
> 
> Sorry, i didn't `git add' in my local copy.
> 
> Done, but in the PR slipped in other changes too (as i view the diff from githun) and
> i don't know if it is normal, so please check first before apply.

By the way. I got this message too, but i couldn't figure out from the link in the
email the reason why. So how you know?

That's what i got:

View results: https://github.com/Jason2605/Dictu/commit/a6dee695944933a71f17d2507778c099a844f1a5/checks?check_suite_id=493007932

Jobs:
  * e7d24538-fce0-54fb-5e84-e8eb6c5e44a8 failed (2 annotations)
  * 6f4601d7-c5e5-575c-7583-759cac63d0cf failed (1 annotation)
  * 73142cd7-fe9a-514b-6dbe-77a751ac9475 failed (1 annotation)

**Comment by Jason2605** [comment]
> On Mon, Mar 02, at 06:09 Jason_000 wrote: Thanks for the PR @agathoklisx! Seems as though the CI/CD integration is failing because it's missing the `setCWD.du` test file.
> Sorry, i didn't `git add' in my local copy. Done, but in the PR slipped in other changes too (as i view the diff from githun) and i don't know if it is normal, so please check first before apply.

Yeah, thats because my develop branch doesn't have any of the HTTP changes merged into it yet. We can go ahead and close this PR and continue with the one you have just opened (since it contains all the changes here). Worth noting however, once you've opened a PR you can still commit changes to the branch and the changes will be reflected in the pull request. So for example, you could `git add` the tests to your `system_setcwd` branch and they would appear in this PR!

> On Mon, Mar 02, at 06:20 agathoklisx wrote: On Mon, Mar 02, at 06:09 Jason_000 wrote: > Thanks for the PR @agathoklisx! Seems as though the CI/CD integration is failing because it's missing the `setCWD.du` test file. Sorry, i didn't `git add' in my local copy. Done, but in the PR slipped in other changes too (as i view the diff from githun) and i don't know if it is normal, so please check first before apply.
> By the way. I got this message too, but i couldn't figure out from the link in the email the reason why. So how you know? That's what i got: View results: https://github.com/Jason2605/Dictu/commit/a6dee695944933a71f17d2507778c099a844f1a5/checks?check_suite_id=493007932 Jobs: * e7d24538-fce0-54fb-5e84-e8eb6c5e44a8 failed (2 annotations) * 6f4601d7-c5e5-575c-7583-759cac63d0cf failed (1 annotation) * 73142cd7-fe9a-514b-6dbe-77a751ac9475 failed (1 annotation)

That is just telling you that the CI/CD process failed, the link should open up to a GitHub actions page telling you why it failed (missing test file).

**Comment by agathoklisx** [comment]
On Mon, Mar 02, at 08:28 Jason_000 wrote:
> That is just telling you that the CI/CD process failed, the link should open up to a GitHub actions page telling you why it failed (missing test file).

Ah so, it was in the raw logs, i missed that, sorry and thanks.

**Comment by Jason2605** [comment]
> On Mon, Mar 02, at 08:28 Jason_000 wrote: That is just telling you that the CI/CD process failed, the link should open up to a GitHub actions page telling you why it failed (missing test file).
> Ah so, it was in the raw logs, i missed that, sorry and thanks.

No worries! Gonna close this PR as #148 supersedes it.

**Comment by Jason2605** [comment]
My bad actually, i see the new PR purely has the additional tests and not the functionality - reopening this PR.

File: c/optionals/system.c [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+        runtimeError("setcwd() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_STRING(args[0])) {
+        runtimeError("setcwd() argument must be a string");
+        return EMPTY_VAL;
+    }
+
+    char *dir = AS_STRING(args[0])->chars;
```
Comment (Jason2605) [inline c/optionals/system.c]: For future reference, there is a macro that "exchanges" a string object (ObjString struct) into a char pointer (char *) `AS_CSTRING` found here: https://github.com/Jason2605/Dictu/blob/develop/c/object.h#L37

**Pull request merged**